Add CWS security profiles (#17084)
DataDog/documentation: content/en/security/cloud_workload_security/security_profiles.md

---
title: CWS Workload Security Profiles
kind: documentation
further_reading:
  - link: "/security/cloud_workload_security/setup"
    tag: "Documentation"
    text: "Setting Up CWS"
  - link: "/security/cloud_workload_security/workload_security_rules"
    tag: "Documentation"
    text: "Managing Cloud Workload Security Rules"
---

Workload Security Profiles provide a baseline of expected workload activity via a behavioral learning model that helps identify potential threats or misconfigurations. This insight can be used when investigating security alerts, including [generating suppression suggestions](#suppress-signals-based-on-suggestions) for known, acceptable workload behavior, as well as identifying previously unseen, anomalous behavior.

## Compatibility

For Workload Security Profiles to be compatible with your Datadog configuration, you must have [CWS enabled][1], use Datadog Agent 7.44 or later, and have activity dumps enabled. Workload Security Profiles are enabled by default and support containerized workloads only.

## How Workload Security Profiles are generated

> Workload Security Profiles are generated when [activity snapshots](#activity-snapshots) from containers sharing a common image name and image tag (version) are merged together to form a security profile. A different security profile is generated for each version of a container image. As a result, one profile is generated per `{ image-name, image-version}` tag combination.

Author: Actually, now that I see your feedback here, I think we should just change this to "image version tag".

> Each security profile acts as a [behavior model](#behavior-learning-model) for a given workload image to identify known, acceptable behavior that is expected to be consistent across all containers versus previously unseen, anomalous behavior that may be unique to a single container or small number of containers.

Reviewer: ```suggestion
Each security profile acts as a [behavior model](#behavior-learning-model) for a given workload image to identify known, acceptable behavior. The security profile uses the behavior model as a template for expected behavior that is consistent across all containers.  
```
Is this sentence a comparison? Before this product, behavior models would group on anomalous behavior? 
`previously, unseen, anomalous behavior that may be unique to a single container or small number of containers.`

Author: It's a comparison, but I see your point. It's more concise and accurate without the comparison.

### Activity snapshots

> Activity snapshots are the building blocks of Workload Security Profiles. They provide a reasonably accurate representation of expected workload behavior. Each snapshot captures kernel-level activity on a container, including process, network, and file access details. This information is collected by the Agent in 30 minute intervals and sent to the Datadog backend.

Reviewer: This is just my preference not to clarify on how accurate something performs.
```suggestion
Activity snapshots are the building blocks of Workload Security Profiles. They provide a representation of expected workload behavior. Each snapshot captures kernel-level activity on a container, including process, network, and file access details. This information is collected by the Agent in 30 minute intervals and sent to the Datadog backend.
```

> When you configure the Agent and enable activity dumps, it automatically starts capturing snapshots for any running cgroups, including those already running when you start the Agent. It also captures snapshots for any new containers in your cloud native environments that use the Linux kernel control groups or cgroups.

Reviewer: Is there another configuration reference for enabling activity dumps? Or is that something that is known by the user? I noticed it's included in the Compatibility section, but wasn't sure if I as a user should already know what this is or not?

Author: There will be, but I'm still waiting on the specifics from Product.

Each Agent can profile multiple containers on the host system simultaneously. To minimize performance overhead, no more than five containers are profiled at any given time.

### Behavior learning model

> A Workload Security Profile is a behavior model for a specific workload image. Capturing and comparing activity snapshots from individual containers enables each security profile to provide a model of the aggregate workload behavior of similar containers. This representation results in a more precise behavior model for a specific workload as more containers are profiled. Since containerized workloads are expected to be immutable, there should be few variations between workloads that share common image name and image version tags.

Reviewer: ```suggestion
A Workload Security Profile is a behavior model for a specific workload image. Capturing and comparing activity snapshots from individual containers enables each security profile to provide a model of the aggregate workload behavior of similar containers. As more containers are profiled, this representation results in a more precise behavior model for a specific workload. Since containerized workloads are expected to be immutable, there should be few variations between workloads that share common image name and image version tags.
```

> A visual representation of the behavior model, including process nodes and file and network activity relationships is shown on the [security profile details page](#explore-security-profiles). The profile visualization displays how consistent or common these relationships are across the various containers that have been profiled.

Reviewer: Is the security profile details page supposed to reference the app or this  #explore-security-profiles section?

Author: The section, but good point. Let me think about adding a link to the app.

#### Model status

> While initial snapshots are being captured and merged into the Workload Security Profile, the profile will be shown with a status of **learning**. After a threshold has been reached in terms of the number of activity snapshots merged or amount of time elapsed since the first snapshot, the security profile status will transition to a **stable** state.

Reviewer: What's the threshold for stable state, or is that determined by the user?

Author: I actually have the same question out to Product right now. I agree, we definitely need to be more specific here.

{{< img src="security/cws/security_profiles/security-profile.png" alt="CWS security profile details page" width="100%">}}

Select a process node to view additional details on the side panel. This includes a commonality score for the process based on how common the files and requests in the process activity are across all workload instances monitored.

Security profile details also appear in the signal panel in the [Security Signals Explorer][3]. You can use this information to help distinguish between potential threats and normal workload behavior. If the security profile matches known workload behavior, a [suppression suggestion](#suppress-signals-based-on-suggestions) is displayed.

{{< img src="security/cws/security_profiles/signal-security-profile.png" alt="Cloud Workload Security security profile page" width="80%">}}

### Suppress signals based on suggestions

If a signal matches known workload behavior for a security profile, a suppression suggestion is displayed on the security signal side panel. You can view the corresponding security profile and the process triggering the signal before choosing whether to accept the suggestion.

> To accept the suggestion, click **Suppress Signals**, then click **Add Suppression to Rule**. This creates a suppression query for the rule that excludes the process from triggering future security signals.

Reviewer: ```suggestion
To accept the suggestion, click **Suppress Signals**, then click **Add Suppression to Rule**. This creates a suppression query for the rule which prevents the process from triggering future security signals.
```
